src/main/java/edu/ucsb/cs156/rec/entities/RecommendationRequest.java

```diff
@@ -0,0 +1,38 @@
+package edu.ucsb.cs156.rec.entities;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+
+/**
+ * This is a JPA entity that represents a RecommendationRequest, i.e. an entry
+ * that comes from the RecommendationRequest API for requests from students.
+ */
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Entity(name = "recommendationrequest")
+public class RecommendationRequest {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+  private String status; 
+
+  private String professorName;
+  private String professorEmail;
```

Comment:
Instead of storing professorName and professorEmail, store professorId, which is a user id from the users table.

Better yet, use this:

```
  @ManyToOne
  @JoinColumn(name = "professor_id")
  private User professor;
```  

This will give you access in a RecommendationRequest object `r` to:
```java
  r. getProfessor().getName();
  r.getProfessor().getEmail();
 // etc...
```

See: https://ucsb-cs156.github.io/topics/spring_boot/spring_boot_database.html#special-case-records-owned-by-a-user-an-intro-to-foreign-keys for more details

Reply:
Fixed